### DISCUSSION TITLE
BETA

### DISCUSSION THREAD

**Discussion by jsdpr15** [pawelsalawa/sqlitestudio#3953] [Q&A]
Sir

Any Beta Release is available of Version 3.2.2 ,
if available please provide Sir

Thanks

**Comment by pawelsalawa** [comment]
I don't practice beta versions anymore. Also there will be no 3.2.2, but already 3.3.0 (see news on the homepage https://sqlitestudio.pl/news/).

**Reply by jsdpr15** [reply]
but sir there were only 3.2.1 version's installer available so how to use 3.3.0

**Reply by pawelsalawa** [reply]
There will be 3.3.0 released soon. Maybe a month, maybe two, maybe three...

**Reply by tiromansev** [reply]
what about night builds?

**Reply by jsdpr15** [reply]
night builds ??

**Reply by tiromansev** [reply]
> night builds ??

build from current git version

**Reply by pawelsalawa** [reply]
Not bad idea, but it requires action setup. I may do it, but I don't want to waste my time on it at this very moment, as I'm putting all my effort to release 3.3.0 ASAP. Maybe sometime during the next few weeks I will have opportunity to do it.

**Reply by pawelsalawa** [reply]
I have fired the build actions for current work in progress (3.3.0). Be aware that I **did not test it**. It may or may not work as expected.

Here are build artifacts:
* Linux: https://github.com/pawelsalawa/sqlitestudio/actions/runs/548327985
* Windows: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545115641
* MaxOS: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545053496

The artifacts are accessible only for GitHub logged users (it's just how GitHub actions handle artifacts). Also they will be present only for next 90 days, after with GitHub will delete them.

**Reply by tiromansev** [reply]
> I have fired the build actions for current work in progress (3.3.0). Be aware that I **did not test it**. It may or may not work as expected.
> 
> Here are build artifacts:
> 
> * Linux: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545891886
> * Windows: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545115641
> * MaxOS: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545053496
> 
> The artifacts are accessible only for GitHub logged users (it's just how GitHub actions handle artifacts). Also they will be present only for next 90 days, after with GitHub will delete them.

thank you for your answer, but i have error when try to start this build

> qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
>This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this >problem.

>Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite->egl, wayland-xcomposite-glx, xcb.

>Aborted (core dumped)

can you help me to understand what wrong, i am using Fedora 33 as main system

**Reply by pawelsalawa** [reply]
Go to `SQLiteStudio/platforms` folder and execute `ldd libqxcb.so`. Tell me what it shows.

**Reply by tiromansev** [reply]
> Go to `SQLiteStudio/platforms` folder and execute `ldd libqxcb.so`. Tell me what it shows.

> ./libqxcb.so: /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libQt5Core.so.5: version `Qt_5.15.2_PRIVATE_API' not found (required by /lib64/libQt5DBus.so.5)
	linux-vdso.so.1 (0x00007ffd11749000)
	libQt5XcbQpa.so.5 => /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libQt5XcbQpa.so.5 (0x00007fd372442000)
	libfontconfig.so.1 => /lib64/libfontconfig.so.1 (0x00007fd3723d7000)
	libfreetype.so.6 => /lib64/libfreetype.so.6 (0x00007fd372314000)
	libz.so.1 => /lib64/libz.so.1 (0x00007fd3722fa000)
	libQt5Gui.so.5 => /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libQt5Gui.so.5 (0x00007fd3719c9000)
	libQt5DBus.so.5 => /lib64/libQt5DBus.so.5 (0x00007fd37193c000)
	libQt5Core.so.5 => /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libQt5Core.so.5 (0x00007fd371144000)
	libGL.so.1 => /lib64/libGL.so.1 (0x00007fd3710bd000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fd37109b000)
	libX11-xcb.so.1 => /lib64/libX11-xcb.so.1 (0x00007fd371096000)
	libxcb-icccm.so.4 => /lib64/libxcb-icccm.so.4 (0x00007fd37108f000)
	libxcb-image.so.0 => /lib64/libxcb-image.so.0 (0x00007fd371089000)
	libxcb-shm.so.0 => /lib64/libxcb-shm.so.0 (0x00007fd371082000)
	libxcb-util.so.1 => /lib64/libxcb-util.so.1 (0x00007fd37107a000)
	libxcb-keysyms.so.1 => /lib64/libxcb-keysyms.so.1 (0x00007fd371075000)
	libxcb-randr.so.0 => /lib64/libxcb-randr.so.0 (0x00007fd371063000)
	libxcb-render-util.so.0 => /lib64/libxcb-render-util.so.0 (0x00007fd37105c000)
	libxcb-render.so.0 => /lib64/libxcb-render.so.0 (0x00007fd37104d000)
	libxcb-shape.so.0 => /lib64/libxcb-shape.so.0 (0x00007fd371046000)
	libxcb-sync.so.1 => /lib64/libxcb-sync.so.1 (0x00007fd37103c000)
	libxcb-xfixes.so.0 => /lib64/libxcb-xfixes.so.0 (0x00007fd371032000)
	libxcb-xinerama.so.0 => /lib64/libxcb-xinerama.so.0 (0x00007fd37102d000)
	libxcb-xkb.so.1 => /lib64/libxcb-xkb.so.1 (0x00007fd371010000)
	libxcb.so.1 => /lib64/libxcb.so.1 (0x00007fd370fe6000)
	libXext.so.6 => /lib64/libXext.so.6 (0x00007fd370fcf000)
	libX11.so.6 => /lib64/libX11.so.6 (0x00007fd370e88000)
	libxkbcommon-x11.so.0 => /lib64/libxkbcommon-x11.so.0 (0x00007fd370e7e000)
	libxkbcommon.so.0 => /lib64/libxkbcommon.so.0 (0x00007fd370e39000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fd370e32000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007fd370c4a000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fd370b02000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007fd370ae7000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fd37091c000)
	libgthread-2.0.so.0 => /lib64/libgthread-2.0.so.0 (0x00007fd370917000)
	libglib-2.0.so.0 => /lib64/libglib-2.0.so.0 (0x00007fd3707e5000)
	libxml2.so.2 => /lib64/libxml2.so.2 (0x00007fd37065a000)
	libbz2.so.1 => /lib64/libbz2.so.1 (0x00007fd370645000)
	libpng16.so.16 => /lib64/libpng16.so.16 (0x00007fd37060c000)
	libbrotlidec.so.1 => /lib64/libbrotlidec.so.1 (0x00007fd3705fe000)
	libdbus-1.so.3 => /lib64/libdbus-1.so.3 (0x00007fd3705ad000)
	libicui18n.so.56 => /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libicui18n.so.56 (0x00007fd370114000)
	libicuuc.so.56 => /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libicuuc.so.56 (0x00007fd36fd5a000)
	libicudata.so.56 => /home/ronin/Downloads/sqlitestudio-3.3.0/SQLiteStudio/platforms/./../lib/libicudata.so.56 (0x00007fd36e377000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fd3729a1000)
	libGLX.so.0 => /lib64/libGLX.so.0 (0x00007fd36e345000)
	libGLdispatch.so.0 => /lib64/libGLdispatch.so.0 (0x00007fd36e28d000)
	libXau.so.6 => /lib64/libXau.so.6 (0x00007fd36e288000)
	libpcre.so.1 => /lib64/libpcre.so.1 (0x00007fd36e20d000)
	liblzma.so.5 => /lib64/liblzma.so.5 (0x00007fd36e1e1000)
	libbrotlicommon.so.1 => /lib64/libbrotlicommon.so.1 (0x00007fd36e1be000)
	libsystemd.so.0 => /lib64/libsystemd.so.0 (0x00007fd36e0fe000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fd36e0f3000)
	libzstd.so.1 => /lib64/libzstd.so.1 (0x00007fd36e01c000)
	liblz4.so.1 => /lib64/liblz4.so.1 (0x00007fd36dffe000)
	libgcrypt.so.20 => /lib64/libgcrypt.so.20 (0x00007fd36ded9000)
	libgpg-error.so.0 => /lib64/libgpg-error.so.0 (0x00007fd36deb4000)

**Reply by pawelsalawa** [reply]
Okay, so this `libQt5Core.so.5: version `Qt_5.15.2_PRIVATE_API' not found (required by /lib64/libQt5DBus.so.5)` is the problem.
I will update build script and re-run the workflow. I'll keep you updated.

**Reply by pawelsalawa** [reply]
I've updated the link in my post above to point to new build.

**Reply by jsdpr15** [reply]
thanks sir for 3.3.0 build, i didn't get any error or not found any issue,
but get an error when i add any database and closed app after reopening app it shows errors that db doesn't exist

**Reply by pawelsalawa** [reply]
If you right-click on this erroneous database and pick "Edit database", what do you see in the dialog? Is the driver selected correctly? Is the file path set correctly?

**Reply by jsdpr15** [reply]
yes right driver selected and path is right and same

**Reply by pawelsalawa** [reply]
Sooo... while you have this Db Dialog open, what happens if you click "Test connection" button at the bottom-left corner? Does it result with green or red indicator?

**Reply by jsdpr15** [reply]
Nothing happens
I think there were issue in installed version so that affect in 3.3.0

**Reply by pawelsalawa** [reply]
Are you saying that you click this:
![image](https://user-images.githubusercontent.com/481846/107770657-0c7d4880-6d3a-11eb-8549-a5b910892d13.png)
and nothing appears here?
![image](https://user-images.githubusercontent.com/481846/107770701-1c952800-6d3a-11eb-85fc-6ac541bc23f9.png)

**Reply by jsdpr15** [reply]
sir did you create any documentation for creating view in studio

**Reply by pawelsalawa** [reply]
Long time ago - when I was creating the User Manual - I planned to do so, but I have not done it yet.
Although creating a view shouldn't really be hard thing to do. You have a button "Add view" and there you type `SELECT` query to execute for this view. Finally commit it to create the view.

**Reply by jsdpr15** [reply]
OK sir
Thanks

**Reply by jsdpr15** [reply]
Sir,
Is sqlite studio work with letest sqlite3 driver 3.34.1?

**Reply by pawelsalawa** [reply]
If you use the build from the link above - then yes. You can always check which version is SQLiteStudio using by going to menu Help -> About. There you go to "Environment" tab and you will see SQLite version there.

**Reply by jsdpr15** [reply]
Yeah sir got it
Thanks

**Reply by jsdpr15** [reply]
> I have fired the build actions for current work in progress (3.3.0). Be aware that I **did not test it**. It may or may not work as expected.
> 
> Here are build artifacts:
> 
> * Linux: https://github.com/pawelsalawa/sqlitestudio/actions/runs/548327985
> * Windows: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545115641
> * MaxOS: https://github.com/pawelsalawa/sqlitestudio/actions/runs/545053496
> 
> The artifacts are accessible only for GitHub logged users (it's just how GitHub actions handle artifacts). Also they will be present only for next 90 days, after with GitHub will delete them.

Sir,
did you make any changes in this action build??

**Reply by pawelsalawa** [reply]
I have changed link in my original comment (and I have written about it above), but not in your comment.

**Reply by jsdpr15** [reply]
Ok Sir

**Reply by jsdpr15** [reply]
Hello Sir,
i know i am asking an nonsense question
but
how to export database creation code file from already created database file to store for security purpose

**Reply by pawelsalawa** [reply]
Use "export database" function and pick "SQL" format for exporting:
![image](https://user-images.githubusercontent.com/481846/110538490-63e9bb00-8124-11eb-978d-2fb42a425705.png)
![image](https://user-images.githubusercontent.com/481846/110538612-811e8980-8124-11eb-98a0-534e85c138e8.png)
![image](https://user-images.githubusercontent.com/481846/110538676-95628680-8124-11eb-9747-a9cade4a8815.png)
![image](https://user-images.githubusercontent.com/481846/110538780-b7f49f80-8124-11eb-991e-0bf66d3b0d5b.png)

**Reply by jsdpr15** [reply]
Why i am getting this error with an old database file
[00:21:06] Cannot load plugin DbSqliteSystemData. Error details: Cannot load library C:\Program Files\SQLiteStudio\plugins\DbSqliteSystemData.dll: Unknown error 0x000000c1.

**Reply by pawelsalawa** [reply]
This plugin is no longer present in SQLiteStudio 3.3.x. Are you using 3.2.x?

**Reply by jsdpr15** [reply]
no sir

**Reply by jsdpr15** [reply]
Sir 
Is it possible to create or calculate some type of report from 3 separate table and store it into 4th table and it will be automatic 
As if I update any record on 1st table so his count and sum of amount will automatically calculate and shown in 4th table

**Reply by pawelsalawa** [reply]
"Views" are exactly for that. You need to educate yourself a little bit about what are views and how to use them.

**Reply by jsdpr15** [reply]
Sir
 if I update or import any data using CSV file so is this possible that it check data in csv file from data table that available at database and only import that data only whose not available in data table?

**Reply by pawelsalawa** [reply]
This is not supported out of the box, but there are some ways to achieve this.

1. One way is to put "UNIQUE" constraint on the column that determines that the data is present or not (an ID of this or, or maybe combination of few columns - in which case this should be compound UNIQUE constraint, defined at the bottom part of Table Window). Once this is prepared you can open import window and on the last page there is an option to "ignore errors", which will simply skip the row if it causes "UNIQUE" constraint to fail.
2. Another way would be to import to a new, temporary table and then `INSERT INTO target_table SELECT * FROM temporary_table WHERE id NOT IN (SELECT id FROM target_table)`. It also relies on having the `id` column and it also can be enhanced to leverage combination of more columns.

Probably there are even more ways to do it.

**Reply by jsdpr15** [reply]
Sir,

did is possible to store names local language sqlite.
bcoz i try to store that into my database but when i look into database using sqlite studio it shows "??????????"

**Reply by pawelsalawa** [reply]
It depends what charset have you used to store the names in the other application, cause SQLiteStudio reads always UTF-8 and it is standard in SQLite3.

**Reply by jsdpr15** [reply]
Sir,

I Create an View whos showing Total of Amount Month wise

but i want to Know that how to add Grand total of shown in View

and how to convert total Amount in Currency Format??

And Also want to know how to set Text align (i.e. Right, Left, Center e.t.c.) in View

**Reply by pawelsalawa** [reply]
Regarding Totals, Grant Totals - you need to learn how to use aggregate functions in your `SELECT` queries: https://sqlite.org/lang_aggfunc.html#aggfunclist

Regarding currency - I'm not sure what you mean here, but if you want to format a number, you can use `printf()` function: https://sqlite.org/lang_corefunc.html#printf

 An alignment is done automatically by SQLite depending of data type returned by a query. If it's a number, it's right-aligned. Otherwise it's left-aligned.

**Reply by jsdpr15** [reply]
> Regarding Totals, Grant Totals - you need to learn how to use aggregate functions in your `SELECT` queries: https://sqlite.org/lang_aggfunc.html#aggfunclist
> 
> Regarding currency - I'm not sure what you mean here, but if you want to format a number, you can use `printf()` function: https://sqlite.org/lang_corefunc.html#printf
> 
> An alignment is done automatically by SQLite depending of data type returned by a query. If it's a number, it's right-aligned. Otherwise it's left-aligned.

how can you please provide example codes

If possible
Please sir

**Reply by pawelsalawa** [reply]
Dear Janil,

This is not SQLite support forum, but SQLiteStudio forum.
I do help people from time to time with SQL or with SQLite, but mostly I help here people with SQLiteStudio related problems. For learning & asking questions about SQL and SQLite, I recommend you stackoverflow.com, there are millions of people ready to help and here I'm mostly the only one.

**Comment by jsdpr15** [comment]
Sir,

is his possible to save any query whose need to execute again and again(periodically) in SQlite Studio Software

**Reply by pawelsalawa** [reply]
- You can save SQL Editor contents into a file and re-load it later if you need.
- You can also just leave SQL Editor open when you close SQLiteStudio and next time you run SQLiteStudio, it will restore you SQL Editor window, together with the query you had in it.
- You can also find your queries executed in the past by switching to History tab (in SQL Editor window) - then double click on a history entry to put it into your execution editor.

**Reply by jsdpr15** [reply]
sir did you add any option to pin or set an alias to mostly used queries

**Reply by pawelsalawa** [reply]
No, but you still have the other 2 options to choose from.

**Reply by pawelsalawa** [reply]
The code snippets feature is on TODO list.

**Comment by jsdpr15** [comment]
Sir,

Any latest version is available of software

**Reply by pawelsalawa** [reply]
3.3.3 is the latest released version. I'm working on 3.4.0 version.

**Comment by jsdpr15** [comment]
Hello sir
Good day
Is there any option to edit print settings in software, bcoz I try to print some reports whose I created in db(as permanent view) so that will separate in 2 or 3 pages

**Reply by pawelsalawa** [reply]
Only whatever settings you see in the printing dialog. Nothing more.
